server/src/main/java/io/druid/server/emitter/HttpEmitterModule.java

```diff
@@ -83,6 +87,17 @@ public Emitter getEmitter(
     if (sslContext != null) {
       builder.withSslContext(sslContext);
     }
-    return new HttpPostEmitter(config.get(), HttpClientInit.createClient(builder.build(), LifecycleUtils.asMmxLifecycle(lifecycle)), jsonMapper);
+    HttpPostEmitter emitter = new HttpPostEmitter(
+        config.get(),
+        HttpClientInit.createClient(builder.build(), LifecycleUtils.asMmxLifecycle(lifecycle)),
+        jsonMapper
+    );
+    HttpPostEmitterMonitor emitterMonitor = new HttpPostEmitterMonitor(
+        FeedDefiningMonitor.DEFAULT_METRICS_FEED,
+        emitter,
+        ImmutableMap.of()
+    );
+    emitterMonitorProvider.setEmitterMontor(emitterMonitor);
```

Comment:
Just an opinion: should we _inject_ the emitter monitor in rather than creating it in a singleton provider?

Reply:
I tried to do this, but didn't find a way. Emitter is not injected statically, it's "chosen" in runtime. Simple injection leads to "duplicate injection" errors. If monitors are also provided via `@Provider` methods along with emitters, emitters still should dynamically "decide" which monitor to use. But I don't see how this approach is better than the current solution.

Reply:
Yeah, I was thinking about the `@Provider` way.

Anyway, my concern is on the order in which the dependencies are injected, so would there be a situation where `MetricsModule` gets loaded before the emitter `Module`s such that `emitterMonitorProvider` returns `null` because it is not set yet? If this is not an issue, then I guess your method would simpler and more straightforward.

Reply:
This is not possible, because in the method where EmitterMonitorProvider is injected, ServiceEmitter is injected as a parameter as well. ServiceEmitter depends on concrete Emitter. So there is a direct dependency chain and Guice will call one of Emitter providers first.

Reply:
Got it :+1: